### PR TITLE
Understanding the Software Development Life Cycle (SDLC)

### LINKED ISSUE #1269: Understanding the Software Development Life Cycle (SDLC)
### Brief Summary: 
This article provides an overview of the steps taken by developers to develop software and related applications. It also takes readers through the most popular SDLC models and highlights their advantages and disadvantages. 

### Key Takeaways: 
1. What is software development life cycle (SDLC)?
2. Importance of SDLC
3. An Overview of SDLC stages
4. Software development process models (with advantages and disadvantages)
5. Limitations of SDLC
6. Conclusion

### References: 
Please list links to any published content/research that you intend to use to support/guide this article. (If none, please indicate 'N/A'.)

### PR DESCRIPTION
**Pull request by mbaabuones**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1269 .

### CODE REVIEW

**Comment by hectorkambow** [comment]
plagiarism check done

**Review by adrianmurage — CHANGES_REQUESTED** [review]
@mbaabuones 
Great article. I left a few notes, kindly make the necessary changes

File: articles/understanding-software-development-life-cycle/index.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
+In the current digital era, many software applications are being developed to solve various real-life problems. If you are a new developer, you would probably want to know the steps that can be taken to develop a software product. Understanding the software development life cycle (SDLC) helps developers to plan effectively and deliver high-quality software products.
```
[the remark this answers is not in the document]
Comment (mbaabuones) [reply articles/understanding-software-development-life-cycle/index.md]: Noted. Thank you.

File: articles/understanding-software-development-life-cycle/index.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
+In the current digital era, many software applications are being developed to solve various real-life problems. If you are a new developer, you would probably want to know the steps that can be taken to develop a software product. Understanding the software development life cycle (SDLC) helps developers to plan effectively and deliver high-quality software products. 
+
+This article takes you through the software development life cycle, which consists of the main stages of software development. It also provides an overview of the model processes in software development and the limitations of SDLC.
```
[the remark this answers is not in the document]
Comment (mbaabuones) [reply articles/understanding-software-development-life-cycle/index.md]: Done.

File: articles/understanding-software-development-life-cycle/index.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
+In the current digital era, many software applications are being developed to solve various real-life problems. If you are a new developer, you would probably want to know the steps that can be taken to develop a software product. Understanding the software development life cycle (SDLC) helps developers to plan effectively and deliver high-quality software products. 
+
+This article takes you through the software development life cycle, which consists of the main stages of software development. It also provides an overview of the model processes in software development and the limitations of SDLC. 
+
+### What is Software Development Life Cycle (SDLC)?
+A software development life cycle (SDLC) refers to the process followed by software developers to develop a high-quality software product. It can also be termed as a software development process. SDLC provides a framework that outlines the tasks that need to be performed in each phase of software development. Organizations can use SDLC to provide a structure for designing and building applications.
```
[the remark this answers is not in the document]
Comment (mbaabuones) [reply articles/understanding-software-development-life-cycle/index.md]: Done.

File: articles/understanding-software-development-life-cycle/index.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+A software development life cycle (SDLC) refers to the process followed by software developers to develop a high-quality software product. It can also be termed as a software development process. SDLC provides a framework that outlines the tasks that need to be performed in each phase of software development. Organizations can use SDLC to provide a structure for designing and building applications. 
+
+This life cycle provides an effective plan for various activities in software development such as designing, building (developing), and maintaining software applications. It also provides a methodology for enhancing the quality of software applications in an organization. Organizations can choose an effective software development strategy from a variety of SDLC process models. 
+
+SDLC can be used by various people in the organization such as software engineers, developers, and cross-function teams. Developers and software engineers use it to create effective plans and designs. They also apply the various SDLC stages to develop innovative software products. Cross-function teams use SDLC to collaborate across different software development stages.
```
[the remark this answers is not in the document]
Comment (mbaabuones) [reply articles/understanding-software-development-life-cycle/index.md]: Done.

File: articles/understanding-software-development-life-cycle/index.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+  
+These questions should be answered carefully and documented before going to the next phase. 
+
+#### Design
+In this phase, the developers are required to refer to the requirements document to design an appropriate architecture for the software. The designs created by the development team should be documented. The document used for documenting the designs is known as a design document specification (DDS). This document should be reviewed by various stakeholders of the organization. 
+
+The final design is selected based on certain parameters such as risk, modularity, robustness, time constraints, and budget. This design provides the blueprint of the software application being developed. 
+
+#### Software Development
+This is the implementation stage that involves developing the software. Here, the design of the product is transformed into reality by creating some source codes. The design document specification provides a guideline for the generation of the source codes. Various programming languages can be used. These include PHP, Java, C++, Pascal, and C. The language chosen depends on the type of software being developed.
```
[the remark this answers is not in the document]
Comment (mbaabuones) [reply articles/understanding-software-development-life-cycle/index.md]: Done.

File: articles/understanding-software-development-life-cycle/index.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+#### Testing
+In this phase, the software is sent to a testing environment where it is tested by test engineers. Testing is done to establish whether the software functions based on the requirements given in the first phase. If the test engineers discover some defects, the developers fix them. The software is tested further to check whether there are other defects. 
+
+#### Deployment
+After ensuring that all the customer requirements have been met, the developers deploy the software for use. Before actual deployment is done, the software should be documented for future reference. Ownership should also be transferred to the client. The deployment strategy depends on the specifications of the customer. It may be deployed to an entire segment or a limited segment. It can be deployed in a data center or a client premise.
```
[the remark this answers is not in the document]
Comment (mbaabuones) [reply articles/understanding-software-development-life-cycle/index.md]: I have revised this section to provide a deeper understanding of the statement.

**Comment by mbaabuones** [comment]
@adrianmurage I have finished working on the changes.

**Comment by adrianmurage** [comment]
@mbaabuones 
Kindly add a hero image

**Comment by mbaabuones** [comment]
> 
> 
> @mbaabuones
> Kindly add a hero image

@adrianmurage I have added a hero image.

**Comment by adrianmurage** [comment]
@mbaabuones 
As always, great job with this. Keep at it!
@hectorkambow this article is ready for final review.

**Pull request merged**